src/control/traversable/basic.lean

```diff
@@ -90,10 +93,23 @@ lemma preserves_seq :
 lemma preserves_map {α β} (x : α → β) (y : F α) : η (x <$> y) = x <$> η y :=
 by rw [← pure_seq_eq_map, η.preserves_seq]; simp with functor_norm
 
+/-- The (trivial) applicative transformation between identity functors. -/
+def id_transformation : applicative_transformation id id :=
+{ app := λ α, id,
+  preserves_pure' := by simp,
+  preserves_seq' := λ a b x y, by simp }
+
+instance : inhabited (applicative_transformation id id) := ⟨id_transformation⟩
```

Comment:
I'm not a big fan of adding (non-instance) definitions just to satisfy the `inhabited` linter. Can't we use `pure_transformation` for the inhabited instance? It doesn't matter that it is in a different file.

Reply:
Oh, I meant for this to be the identity in the category of applicative transformations, but I messed it up.  Would you mind taking a look at it again @fpvandoorn ?

There's also a composition law, but I probably shouldn't add it with this lint PR, right?
```lean
universe s
variables {H : Type u → Type s} [applicative H] [is_lawful_applicative H]

/-- The composition of applicative transformations. -/
def comp (η : applicative_transformation F G) (η' : applicative_transformation G H) :
  applicative_transformation F H :=
{ app := λ α, η'.app α ∘ η.app α,
  preserves_pure' := λ α x, by simp [η.preserves_pure', η'.preserves_pure'],
  preserves_seq' := λ α β x y, by simp [η.preserves_seq', η'.preserves_seq'] }
```

Reply:
Oh, that's much better, and that should definitely be in the library. You can open a new PR for the composition law.